bin/utils.js

```diff
@@ -228,11 +236,17 @@ const closeConn = (doc, conn) => {
     doc.conns.delete(conn)
     awarenessProtocol.removeAwarenessStates(doc.awareness, Array.from(controlledIds), null)
     if (doc.conns.size === 0 && persistence !== null) {
-      // if persisted, we store state and destroy ydocument
-      persistence.writeState(doc.name, doc).then(() => {
-        doc.destroy()
-      })
-      docs.delete(doc.name)
+      setTimeout(() => {
```

Comment:
Should we clear this timeout if someone connects? Also, if we get that this is called multiple times, we can schedule multiple timers

Reply:
yeah yeah I'm working on timeouts as we speak lol